Source/AliveLibAO/Abe.cpp

```diff
@@ -4455,9 +4456,150 @@ void Abe::State_16_HoistBegin_426E40()
+                    { rect.w, rect.h },
+                    ObjListPlatforms_50766C,
+                    1,
+                    (TCollisionCallBack) &BaseAliveGameObject::OnTrapDoorIntersection_401C10
+                );
+                break;
+            }
+        }
+        return;
+    }
+
+    PullRingRope* pPullRope = GetPullRope_422580();
+    field_160_pRope = pPullRope;
+    if (pPullRope)
+    {
+        if (pPullRope->Pull_454CB0(this))
+        {
+            field_FC_current_motion = eAbeStates::State_69_RingRopePull_4299B0;
+            field_FE_next_state = eAbeStates::State_0_Idle_423520;
+            field_160_pRope->field_C_refCount++;
+            return;
+        }
+        field_160_pRope = nullptr;
+    }
+
+    if (field_B8_vely >= FP_FromInteger(0))
+    {
+        Path_Hoist* pHoist = static_cast<Path_Hoist*>(gMap_507BA8.TLV_Get_At_446260(
+            FP_GetExponent(field_A8_xpos),
+            FP_GetExponent(field_AC_ypos),
+            FP_GetExponent(field_A8_xpos),
+            FP_GetExponent(field_AC_ypos),
+            TlvTypes::Hoist_3
+        ));
+        field_F0_pTlv = pHoist;
+        if (IsSameScaleAsHoist(pHoist, this))
+        {
+            if (IsFacingSameDirectionAsHoist(pHoist, this))
+            {
+                if (pHoist->field_1A_edge_type == Path_Hoist::EdgeType::eBoth)
+                {
+                    if (gMap_507BA8.SetActiveCameraDelayed_444CA0(Map::MapDirections::eMapTop_2, this, -1))
+                    {
+                        PSX_Prevent_Rendering_44FFB0();
+                        field_FC_current_motion = eAbeStates::State_67_ToOffScreenHoist_428C50;
+                        return;
+                    }
+                    field_AC_ypos -= field_BC_sprite_scale * FP_FromInteger(80);
+                    field_D0_pShadow->field_14_flags.Set(Shadow::Flags::eBit1_ShadowAtBottom);
+                }
+
+                field_FC_current_motion = eAbeStates::State_66_LedgeHang_428D90;
+                Environment_SFX_42A220(EnvironmentSfx::eWalkingFootstep_1, 0, 127, this);
+
+
+                if (sCollisions_DArray_504C6C->RayCast_40C410(
+                    field_A8_xpos,
+                    field_AC_ypos - field_BC_sprite_scale * FP_FromInteger(80),
+                    field_A8_xpos,
+                    field_AC_ypos,
+                    &pPathLine,
+                    &hitX,
+                    &hitY,
+                    field_BC_sprite_scale != FP_FromDouble(0.5) ? 1 : 0x10))
+                {
+                    field_AC_ypos = hitY;
+                    field_F4_pLine = pPathLine;
+                    field_B8_vely = FP_FromInteger(0);
+                    if (!field_F8_pLiftPoint)
+                    {
+                        if (pPathLine->field_8_type == 32 || pPathLine->field_8_type == 36)
+                        {
+                            PSX_RECT rect = {};
+                            VGetBoundingRect_418120(&rect, 1);
```

Comment:
dito

Reply:
rebuilt the PR on travis, it now fails because I merged my PR

C:\ar\Source\AliveLibAO\Abe.cpp(4492): error C2248: 'AO::BaseAnimatedWithPhysicsGameObject::VGetBoundingRect_418120': cannot access private member declared in class 'AO::BaseAnimatedWithPhysicsGameObject' [C:\ar\build\Source\AliveLibAO\AliveLibAO.vcxproj]